sprocs/instance_questions_points_exam.sql

```diff
@@ -23,31 +23,40 @@ BEGIN
     SELECT * INTO aq FROM assessment_questions WHERE id = iq.assessment_question_id;
 
     max_points := COALESCE(aq.max_points, 0);
-    highest_submission_score := greatest(submission_score, coalesce(highest_submission_score, 0));
 
-    correct := (submission_score >= 0.5);
+    -- Update points (instance_question will be closed when number_attempts exceeds bound,
+    -- so we don't have to worry about accessing a non-existent entry in points_list_original,
+    -- but use coalesce just to be safe)
```

Comment:
On the line below we don't need `coalesce` around `iq.points` or `iq.points_list_original`. We are assuming that these are non-NULL.

In fact, I think we are assuming everything is non-NULL except for `iq.highest_submission_score`, which starts as NULL when `number_attempts = 0` and is non-NULL thereafter. So I think we only need a `coalesce` in the case of `coalesce(iq.highest_submission_score, 0)`.

Reply:
done